external/js/cothority/src/network/proto.ts

```diff
@@ -251,15 +218,28 @@ export class ServerIdentity extends Message<ServerIdentity> {
     }
 
     /**
-     * Returns websocket version of this.url if set, otherwise converts the server
-     * address to match the websocket format.
+     * Returns websocket URL to connect to this ServerIdentity.
      * @returns the websocket address
      */
     getWebSocketAddress(): string {
-        if (this.url) {
-            return ServerIdentity.urlToWebsocket(this.url);
+        if (this.url !== undefined && this.url !== "") {
+            const url = new URL(this.url, {});
```

Comment:
```javascript
new URL(this.url)
```
the 2nd is optional

Reply:
If `this.url` is relative (it might as there isn't checks for it), it will use `windows.location` to resolve it. Specifing an empty second argument forces the URL to be absolute.